rust/cardano-blockchain-types/src/cip36/key_registration.rs

```diff
@@ -0,0 +1,368 @@
+//! CIP-36 Key Registration 61284.
+//!
+//! Catalyst registration data
+//!
+//! <https://cips.cardano.org/cip/CIP-36>
+//! <https://github.com/cardano-foundation/CIPs/blob/master/CIP-0036/schema.cddl>
+
+use std::collections::HashSet;
+
+use anyhow::Context;
+use ed25519_dalek::VerifyingKey;
+use minicbor::{decode, Decode, Decoder};
+use pallas::ledger::addresses::{Address, ShelleyAddress};
+use strum::FromRepr;
+
+use super::voting_pk::VotingPubKey;
+use crate::utils::decode_helper::{decode_array_len, decode_bytes, decode_helper, decode_map_len};
+
+/// CIP-36 key registration - 61284
+///
+///
+/// ```cddl
+/// key_registration = {
+///     1 : [+delegation] / legacy_key_registration,
+///     2 : $stake_credential,
+///     3 : $payment_address,
+///     4 : $nonce,
+///     ? 5 : $voting_purpose .default 0
+//   }
+/// ```
+#[allow(clippy::module_name_repetitions)]
+#[derive(Clone, Default, Debug)]
+pub struct Cip36KeyRegistration {
```

Comment:
Shouldn't need to be public, should be an inner type of a fully deserialized and validated CIP36 structure.